graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ConfigGenerator.java

```diff
@@ -0,0 +1,304 @@
+        NodeList<Node> fields = connectionConfigNode.fields();
+        tokens.add(authConfigNode);
+        for (Node n : fields) {
+            tokens.add(n);
+        }
+        NodeList<Node> nodeList = createNodeList(tokens);
+        RecordTypeDescriptorNode.RecordTypeDescriptorNodeModifier recordTypeDescriptorNodeModifier =
+                connectionConfigNode.modify().withFields(nodeList);
+        connectionConfigNode = recordTypeDescriptorNodeModifier.apply();
+
+        return createTypeDefinitionNode(((TypeDefinitionNode) node).metadata().get(),
+                createToken(PUBLIC_KEYWORD), createToken(TYPE_KEYWORD), ((TypeDefinitionNode) node).typeName(),
+                connectionConfigNode, createToken(SEMICOLON_TOKEN));
+    }
+
+    /**
+     * Gets the path of the utils.bal template at the time of execution.
+     *
+     * @return Path to utils.bal file in the temporary directory created
+     * @throws IOException When failed to get the templates/utils.bal file from resources
+     */
+    private Path getResourceFilePath() throws IOException {
+        Path path = null;
+        ClassLoader classLoader = getClass().getClassLoader();
+        InputStream inputStream = classLoader.getResourceAsStream("templates/config_types_graphql.bal");
+        if (inputStream != null) {
+            String clientSyntaxTreeString = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+            Path tmpDir = Files.createTempDirectory(".config_types_graphql-tmp" + System.nanoTime());
+            path = tmpDir.resolve("config_types_graphql.bal");
+            try (PrintWriter writer = new PrintWriter(path.toString(), StandardCharsets.UTF_8)) {
+                writer.print(clientSyntaxTreeString);
+            }
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                try {
+                    FileUtils.deleteDirectory(tmpDir.toFile());
+                } catch (IOException ex) {
+                    LOGGER.error("Unable to delete the temporary directory : " + tmpDir, ex);
+                }
+            }));
+        }
+        return path;
+    }
+
+    private List<Node> generateApiKeysConfigRecordFields(AuthConfig authConfig) {
+        List<Node> ballerinaApiKeysConfigRecords = new ArrayList<>();
+        for (String headerName : authConfig.getApiHeaders()) {
+            MetadataNode metadataNode = createMetadataNode(null, createEmptyNodeList());
+            TypeDescriptorNode typeName = createSimpleNameReferenceNode(createToken(STRING_KEYWORD));
+            IdentifierToken apiKeyFieldName =
+                    createIdentifierToken(CodeGeneratorUtils.getValidName(headerName));
+            ballerinaApiKeysConfigRecords.add(
+                    createRecordFieldNode(metadataNode, null, typeName,
+                            apiKeyFieldName, null, createToken(SEMICOLON_TOKEN)));
+        }
+        return ballerinaApiKeysConfigRecords;
+    }
+
+    /**
+     * Generates the utils file content.
```

Comment:
Let's update the documentation comment

Reply:
Updated in https://github.com/ballerina-platform/graphql-tools/commit/d6020522edacade9f7e6537cff02da89fa0b1b21